### DISCUSSION TITLE
Send MP3/audio file to BT sink on trigger

### DISCUSSION THREAD

**Discussion by GSVNoFixedabode** [pschatzmann/ESP32-A2DP#415] [Q&A] [answered]
Could someone please point me to one of the example files (impressive library!) that I could use for the following please?
Standalone (battery) ESP32 with Ultrasonic sensor, sends sound to BT speaker on motion detection
- short sound (5 seconds/MP3), I assume I'll need to convert that to C Array
So code blocks should be simple:
- Ultrasound sending pulses every 1 second (that I can code)
-  -- trigger (return detects distance less than X cm)
-  -- play sound to BT Sink
-  -- pause for X seconds
- Loop back to waiting for detection.
I've found base-generator-a2dp.ino to send a Sine tone, but new to C so struglging a bit.  I need to find the example library that reads in a C array file(?) to put together with the BT sink. 
Many thanks.

**Comment by pschatzmann** [comment, marked as answer]
The Readme of this project contains all the information that you need! If you are new to audio, maybe the introduction to the [AudioTools Wiki](https://github.com/pschatzmann/arduino-audio-tools/wiki) might help. Also read the [Readme of the AudioTools](https://github.com/pschatzmann/arduino-audio-tools). I suggest to use PCM data if possible instead of mp3. Otherwise you will need to do some more reading in the AudioTools...

I suggest to use the MemoryStream from the AudioTools as data source: This should make the implementation quite trivial...

Just learn how to use the callback! In the pause you would just send 0 values...

**Reply by GSVNoFixedabode** [reply]
Thanks for the reply.  I've managed to get the above working using the 'piano8bit.h' file (well half of it, space reasons) and the ESP32-A2DP library, so it'll play the piano notes to a BT speaker whenever somethingcomes within XXcm distance of the sensor.  Took me a few days to realise that the sound file I was using (train whistle sound file converted from MP3 to .h via Audacity 16bit/44100/2channel) was crashing things despite being smaller than the piano sample. 
Are you able to point me to the right library or example file that I'd need to read an MP3 or WAV file from an attached SD card rather than an included .h file please? Do I need to change from ESP32-A2DP library ot the toolkit library, or use both? 
(yes, I AM new to Arduino/C/esp32. Nearest experence has been Thonny Python on Pico so struggling a bit here)

**Comment by pschatzmann** [comment]
Just use https://github.com/pschatzmann/arduino-audio-tools: Read the Readme and the Wiki
